File: app/src/main/java/com/davismiyashiro/expenses/view/managetabs/ChooseTabsInterfaces.java
package com.davismiyashiro.expenses.view.managetabs;

import com.davismiyashiro.expenses.datatypes.Tab;

import java.util.List;

/**
 * Created by Davis on 26/08/2016.
 */
public interface ChooseTabsInterfaces {
    interface View {
        void setProgressIndicator(boolean active);
        void showTabs(List<Tab> tabs);
        void showNoTabs();
        void showAddTab();
        void showTabDetailUi(Tab tab);
    }

    interface UserActionsListener {
        void addTab();
        void loadTabs(boolean forceUpdate);
        void openTab(String tabId);
        void removeTab(String tabId);
    }
}

File: app/src/main/java/com/davismiyashiro/expenses/view/managetabs/ChooseTabsInterfaces.kt
package com.davismiyashiro.expenses.view.managetabs

import com.davismiyashiro.expenses.datatypes.Tab

/**
 * Created by Davis on 26/08/2016.
 */
interface ChooseTabsInterfaces {
    interface View {
        fun setProgressIndicator(active: Boolean)
        fun showTabs(tabs: MutableList<Tab>)
        fun showNoTabs()
        fun showAddTab()
        fun showTabDetailUi(tab: Tab)
    }

    interface UserActionsListener {
        fun addTab()
        fun loadTabs(forceUpdate: Boolean)
        fun openTab(tabId: String)
        fun removeTab(tabId: String)
    }
}
